packages/jaeger-ui/src/components/App/TraceIDSearchInput.tsx

```diff
@@ -36,8 +36,18 @@ class TraceIDSearchInput extends React.PureComponent<Props> {
 
   render() {
     return (
-      <Form layout="horizontal" onSubmit={this.goToTrace} className="TraceIDSearchInput--form">
-        <Input name="idInput" placeholder="Lookup by Trace ID..." prefix={<Icon type="search" />} />
+      <Form
+        data-testid="TraceIDSearchInput--form"
```

Comment:
is `data-testid` preserved in the production output or stipped out?

Reply:
I just read about it, and yes it ships to the production builds.

The official [FAQs](https://testing-library.com/docs/dom-testing-library/faq/) of RTL suggests using this [babel plugin](https://www.npmjs.com/package/babel-plugin-react-remove-properties) for stripping out that in production, if that's an issue

Reply:
let's use that plugin then, it's weird to have useless test properties in prod

Reply:
I agree. Never realized production code needs this much sanity.

Reply:
So I think this specific component is not as critical since there's only one of it, but I want to establish a _sustainable pattern_ that can be used for _all tests_, as some of the components could be as granular as a single span in a trace (and a trace may have 10s of thousands of spans), so having these test properties on those components will have noticeable impact on prod HTML.

Reply:
Absolutely. It might affect performance, as you stated, because each chunk of extra data matters.

I have added the Babel plugin to strip that off.

I manually analyzed the build files, and we were correct, it was having that `"data-testid":"TraceIDSearchInput--form"` attribute.
Took multiple iterations of `yarn build` to finally make the plugin work with vite tooling.